### PR TITLE
fix dag naming convention, general improvements on the dag structures

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

File: src/bobtail/dag.cpp [hunk 1]
```diff
@@ -66,46 +61,113 @@ uint16_t CDagNode::GetNodeScore()
...
+    new_id = base_dag_id;
+    return true;
+}
+
+void CBobtailDagSet::Clear()
+{
+    vdags.clear();
+}
+
+CDagNode* CBobtailDagSet::Find(const uint256 &hash)
```
Comment (bissias) [inline src/bobtail/dag.cpp:164]: For the Find method, I wonder if we will regret obscuring the DAG where it came from? Perhaps it's better to return a `std::pair<int16_t, CDagNode*>`? I don't have strong feelings about this.

File: src/bobtail/dag.cpp [hunk 2]
```diff
@@ -127,40 +189,44 @@ bool CDagForrest::Insert(const CSubBlock &sub_block)
...
-        next_subgraph_id = next_subgraph_id + 1;
+        new_id = vdags.size();
+        vdags.emplace_back(new_id, newNode);
     }
-    newNode->subgraph_id = new_id;
+    newNode->dag_id = new_id;
+    vdags[new_id].Insert(newNode);
     // TODO : should insert to maintain temporal ordering not just emplace_back
-    _dag.emplace_back(newNode);
+    mapAllNodes.emplace(newNode->hash, newNode);
```
Comment (bissias) [inline src/bobtail/dag.cpp:219]: Theoretically mapAllNodes could be a map of maps: `std::map<uint256, <uint16_t, CDagNode*>>`. I don't have strong feelings about this.

File: src/bobtail/dag.cpp [hunk 3]
```diff
@@ -127,40 +189,44 @@ bool CDagForrest::Insert(const CSubBlock &sub_block)
...
+bool CBobtailDagSet::IsTemporallySorted()
 {
     return true;
 }
 
-bool CDagForrest::IsSubgraphValid(std::set<uint256> sgHashes)
+bool CBobtailDagSet::IsSubgraphValid(std::set<uint256> sgHashes)
 {
     // TODO : replace 15 with consensus k variable
     if (sgHashes.size() < 15)
```
Comment (bissias) [inline src/bobtail/dag.cpp]: Is "15" holding the place of something other than `BOBTAIL_K`?

File: src/bobtail/dag.cpp [hunk 4]
```diff
@@ -127,40 +189,44 @@ bool CDagForrest::Insert(const CSubBlock &sub_block)
...
 }
 
-bool CDagForrest::IsTemporallySorted()
+bool CBobtailDagSet::IsTemporallySorted()
 {
     return true;
 }
 
-bool CDagForrest::IsSubgraphValid(std::set<uint256> sgHashes)
+bool CBobtailDagSet::IsSubgraphValid(std::set<uint256> sgHashes)
```
Comment (bissias) [inline src/bobtail/dag.cpp]: I got the sense that you were replacing "subgraph" with "dag" thought? Is that the case? If so, shouldn't this be `IsDagValid`?

File: src/bobtail/dag.cpp (hunk 2 above)
Comment (Greg-Griffith) [reply src/bobtail/dag.cpp:219]: its supposed to be an unordered map of all nodes in every dag. you can get the id of the dag from inside the node itself if needed

Comment (bissias) [reply src/bobtail/dag.cpp:219]: I see. So would that be `bobtailDagSet.vdags[bobtailDagSet.Find(hash)->dag_id]`?

File: src/bobtail/dag.cpp (hunk 3 above)
Comment (Greg-Griffith) [reply src/bobtail/dag.cpp]: method was removed

File: src/bobtail/dag.cpp (hunk 4 above)
Comment (Greg-Griffith) [reply src/bobtail/dag.cpp]: method was removed

File: src/bobtail/dag.cpp (hunk 2 above)
Comment (Greg-Griffith) [reply src/bobtail/dag.cpp:219]: to get the dag it is in, yes.

File: src/bobtail/dag.cpp (hunk 1 above)
Comment (Greg-Griffith) [reply src/bobtail/dag.cpp:164]: is searching for a dag that a node belongs to something you find yourself needing in other areas of the code?

Comment (bissias) [reply src/bobtail/dag.cpp:164]: No, not presently. Just trying to think ahead. I think we should leave things as you have implemented them for now.

**Pull request merged**